x-pack/plugins/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -148,12 +208,74 @@ export function Insight({ messages, title, dataTestSubj }: InsightProps) {
 
   let children: React.ReactNode = null;
 
-  if (hasOpened && connectors.selectedConnector) {
+  if (
+    connectors.selectedConnector &&
+    ((!isInsightOpen && hasOpened) || (isInsightOpen && !isEditingPrompt))
+  ) {
     children = (
-      <ChatContent
-        title={title}
-        initialMessages={messages}
-        connectorId={connectors.selectedConnector}
+      <>
+        {updatedPrompt ? (
+          <>
+            <EuiFlexGroup alignItems="center" gutterSize="none">
+              <EuiFlexItem grow={false}>
+                {i18n.translate('xpack.observabilityAiAssistant.insightModifiedPrompt', {
+                  defaultMessage: 'This insight has been modified.',
+                })}
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  data-test-subj="observabilityAiAssistantInsightResetDefaultPrompt"
+                  onClick={() => {
+                    setUpdatedPrompt('');
+                    setHasOpened(false);
+                    setInsightOpen(false);
+                    setInitialMessages(messages);
+                  }}
+                >
+                  {i18n.translate('xpack.observabilityAiAssistant.resetDefaultPrompt', {
+                    defaultMessage: 'Reset to default',
+                  })}
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+
+            <EuiHorizontalRule size="full" margin="s" />
+          </>
+        ) : null}
+
+        <ChatContent
+          title={title}
+          initialMessages={initialMessages}
+          connectorId={connectors.selectedConnector}
+        />
+      </>
+    );
+  } else if (isEditingPrompt) {
+    children = (
+      <PromptEdit
+        initialPrompt={
+          updatedPrompt ||
+          last(messages.filter((msg) => msg.message.role === MessageRole.User))?.message.content ||
+          ''
+        }
+        onSend={(newPrompt) => {
```

Comment:
I would personally have these callbacks defined as a variable, i.e. 

```
const handleSend = () => {
  ...
}

...

<PromptEdit
  ...
  onSend={handleSend}
  ...
/>
```

For the reason of not creating a new function on every render. But it's not super important.

Reply:
fixed in https://github.com/elastic/kibana/pull/176086/commits/3ef74e8b4784383775bfc0b38631a2bc18db8e6d